### PR TITLE
feature(git-commands): Git commands and git rebase

### PR DESCRIPTION
**Pull request by michaelachrisco**
## Changes
1. Adds git commands README
2. Adds `git stash` with example/references

## Approach
By explaining git stash and the concepts behind it, I am hoping to help other developers be more productive when a clean index/working directory is needed as well as switching code between branches for context switching purposes. It also allows the developer to maintain feature specific branches by allowing code changes to not be "lost" when switching between branches.

At the moment, I plan on being as specific as possible with each of these examples, so I will try to be as atomic as possible with these small PRs.

## Learning
I had to look up a large amount of resources as there is more documentation on rebasing than git stashing.
I'm still looking for a good video tutorial to accommodate visual learning but at the moment, I cant find a great resource. Any help would be appreciated!

References: #192

### CODE REVIEW

File: standards/git-commands.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+$ git stash pop
+On branch mac-test-branch
+Changes to be committed:
+  (use "git restore --staged <file>..." to unstage)
+	new file:   standards/git-commands.md
+
+Dropped refs/stash@{0} (1a5b247bdd51df733d9ab2e1d1aadc270eded2ce)
+```
+
+A good visual representation can be seen in Gitkrakens documentation: https://support.gitkraken.com/working-with-commits/stashing/
```
Comment (mwallert) [inline standards/git-commands.md:34]: Might be worthwhile to markdownify this link: [Gitkraken Stashing](https://support.gitkraken.com/working-with-commits/stashing/)

Comment (michaelachrisco) [reply standards/git-commands.md]: Fixed via: 35d5bd28c7bdad68ec2a20ec8da4708471e95610

Thanks!

File: standards/git-commands.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Git Commands
+Git is a distributed version-control system for tracking changes in source code during software development. It is designed for coordinating work among programmers, but it can be used to track changes in any set of files. Its goals include speed, data integrity, and support for distributed, non-linear workflows.
+
+At Shift3, we use git for a majority of our software projects.
+
+## Shift3 git specific resources
```
Comment (coreyshuman) [inline standards/git-commands.md:6]: I think we should be consistent about how we capitalize words in titles. Do we want to capitalize all words, as in `Git Commands` above and `Git Stash` below? Not high priority just something I noticed.

File: standards/git-commands.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+2. [Branching](branching.md)
+3. [Code Versioning](code-versioning.md)
+
+### Lesser known git commands
+
+#### Git Stash
+
+Git Official Documentation: https://git-scm.com/docs/git-stash
+
+Git Stash is used to record the current state of the working directory and index (your code within your current branch) but want to go back to a clean working directory. In order to "stash" your code away for future use, first `git add` the files you want to stash away but *do not use git commit*. You use the `git stash push` command to create your stash. You can switch your branches and then apply the same code to the new branch with `git stash pop`.
```
Comment (coreyshuman) [inline standards/git-commands.md:17]: This first sentence is a bit confusing and took me a few reads to understand. Perhaps rearranging and splitting into two sentence may convey the meaning a bit better? Something like:
`Git Stash is useful when you want to go back to a clean working directory but don't want to discard your current changes. It works by locally storing the current state of the working directory and index (the code within the current branch).`

File: standards/git-commands.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+### Lesser known git commands
+
+#### Git Stash
+
+Git Official Documentation: https://git-scm.com/docs/git-stash
+
+Git Stash is used to record the current state of the working directory and index (your code within your current branch) but want to go back to a clean working directory. In order to "stash" your code away for future use, first `git add` the files you want to stash away but *do not use git commit*. You use the `git stash push` command to create your stash. You can switch your branches and then apply the same code to the new branch with `git stash pop`. 
+
+In the below example, we are pushing this readme into our `foo-999-big-feature` branch which leaves us with a clean working directory/index within our origional branch `mac-192-lesser-known-git`.
```
Comment (coreyshuman) [inline standards/git-commands.md:19]: Spelling of `original`

File: standards/git-commands.md (hunk 2 above)
Comment (michaelachrisco) [reply standards/git-commands.md]: Nice catch, this should fix the titles: 46a852e9612e94dc7955cb2810dccd0e0c1474c1

File: standards/git-commands.md (hunk 3 above)
Comment (michaelachrisco) [reply standards/git-commands.md:17]: I think I took some liberties with the official git documentation and may have made it a bit more confusing. Ill go ahead with the changes.

Comment (michaelachrisco) [reply standards/git-commands.md]: Should be fixed via: dd8bf571fbb14c9765797c3c712d90912ab3cc3a

File: standards/git-commands.md (hunk 4 above)
Comment (michaelachrisco) [reply standards/git-commands.md]: Doh!

Should be fixed via: dd8bf571fbb14c9765797c3c712d90912ab3cc3a

**Comment by michaelachrisco** [comment]
Thanks for the review @mwallert and @coreyshuman. 

I believe I have addressed the issues. Spelling is my curse so thank you both for reviewing!

**Pull request merged**